### PR TITLE
Transform `Environment` class into Singleton

### PR DESCRIPTION
**Pull request by neomatamune**
Removed all uses of the decorator `pass_environment`

### CODE REVIEW

File: Babylon/utils/environment.py [hunk 1]
```diff
@@ -1,26 +1,45 @@
 import os
```
Comment (o-reo) [inline Babylon/utils/environment.py:1]: I thought you would use a module variable
```python
class Environment():
    pass

environment = Environment()
```
and then 
```python
from ..utils.environment import environment
```
But a real singleton is another option !

Comment (neomatamune) [reply Babylon/utils/environment.py:1]: I never did a Singleton before so it was an occasion to do it ^^

File: Babylon/groups/config/__init__.py [hunk 2]
```diff
@@ -4,15 +4,14 @@
...
 
 
 @group()
-@pass_environment
 @pass_context
-def config(ctx: Context, env):
+def config(ctx: Context):
     """Group made to work on the config"""
-    ctx.obj = env.configuration
+    ctx.obj = Environment().configuration
```
Comment (o-reo) [inline Babylon/groups/config/__init__.py:14]: I think it's unnecessary ?

File: Babylon/groups/working_dir/__init__.py [hunk 3]
```diff
@@ -3,15 +3,14 @@
...
 
 
 @group()
-@pass_environment
 @pass_context
-def working_dir(ctx, environment):
+def working_dir(ctx):
     """Command group handling working directory information"""
-    ctx.obj = environment.working_dir
+    ctx.obj = Environment().working_dir
```
Comment (o-reo) [inline Babylon/groups/working_dir/__init__.py:13]: Probably unnecessary !

**Review by o-reo — APPROVED** [review]
:+1: :100:

File: Babylon/groups/working_dir/__init__.py (hunk 3 above)
Comment (neomatamune) [reply Babylon/groups/working_dir/__init__.py:13]: Yeah probably forgot to remove those ^^'

**Pull request merged**